plugins/destination/mssql/queries/values.go

```diff
@@ -0,0 +1,114 @@
+package queries
+
+import (
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v2/types"
+	mssql "github.com/microsoft/go-mssqldb"
+	"golang.org/x/exp/slices"
+)
+
+func GetRows(reader array.RecordReader) ([][]any, error) {
+	var rows [][]any
+	for reader.Next() {
+		r, err := getRecordRows(reader.Record())
+		if err != nil {
+			return nil, err
+		}
+		rows = append(rows, r...)
+	}
+	return slices.Clip(rows), nil
+}
+
+func prealloc(rows, cols int64) [][]any {
+	result := make([][]any, rows)
+	for i := range result {
+		result[i] = make([]any, cols)
+	}
+	return result
+}
+
+func getRecordRows(record arrow.Record) ([][]any, error) {
+	rows := prealloc(record.NumRows(), record.NumCols())
+	var err error
+
+	for row := range rows {
+		for idx, col := range record.Columns() {
+			rows[row][idx], err = getColValue(col, row)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return rows, nil
+}
+
+func getColValue(arr arrow.Array, idx int) (any, error) {
+	if arr.IsNull(idx) {
+		return nil, nil
+	}
+
+	switch arr := arr.(type) {
+	case *array.Boolean:
+		return ptr(arr.Value(idx)), nil
+
+	case *array.Uint8:
+		return ptr(arr.Value(idx)), nil
+	case *array.Uint16:
+		return ptr(int16(arr.Value(idx))), nil // as we map those to the signed types for now
+	case *array.Uint32:
+		return ptr(int32(arr.Value(idx))), nil // as we map those to the signed types for now
+	case *array.Uint64:
+		return ptr(int64(arr.Value(idx))), nil // as we map those to the signed types for now
```

Comment:
I've discussed with @yevgenypats and it was decided to do the simple signed conversion for now.

Reply:
Updated the code in 6558380.
uint64 is stored as string for now.

Reply:
So Im not sure there is a dataloss here. In PostgreSQL there is also no such thing as unsigned because unsigned is mostly how something is read then how it is stored. so let's say we have `Uint16` `65535` it will be stored as "-32766" but if you know it's a Uint then you can always convert it back but there is no dataloss. I don't think we need to store `uint64` as string but rather store it as int64.

to make things more friendly we can convert `uint16` to `int32` but it is mostly for user friendliness rather then dataloss (as there is not dataloss because uint16 and int16 are stored the same way in memory).

Reply:
I think the main concern in how this data is handled inside the database.
I think the current upscale is OK, apart form `uint64`:
* `uint8` -> `tinyint`
* `uint16` -> `int32` -> `int`
* `uint32` -> `int64` -> `bigint`
* `int8` -> `int16` -> `smallint`
* `int16` -> `smallint`
* `int32` -> `int`
* `int64` -> `bigint`

I think it'll be a better user experience to convert `uint64` to `int64` and hope for the best (we can always fix this later after a report from user).
@hermanschaaf @yevgenypats WDYT?

Reply:
anyway, don't have a strong opinion on `uint64` we can do either and then change.

Reply:
Just in case, the option with uint64 as int64 is implemented in 768aff6

Reply:
Yeah, I think for the sake of user experience `uint64` as `int64` is better for now, we don't really have a good alternative.